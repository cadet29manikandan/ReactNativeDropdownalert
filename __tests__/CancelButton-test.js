import React from 'react';
import CancelButton from '../CancelButton';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('CancelButton', () => {
  test('renders', () => {
    const wrapper = shallow(<CancelButton />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
  test('test onPress to be defined', () => {
    const onCancel = () => {
      console.log('Cancelled');
    };
    const wrapper = shallow(<CancelButton onPress={onCancel} />);
    expect(wrapper.prop('onPress')).toEqual(onCancel);
    expect(wrapper.props().onPress).toBeDefined();
    CancelButton.defaultProps.onPress();
  });
});
