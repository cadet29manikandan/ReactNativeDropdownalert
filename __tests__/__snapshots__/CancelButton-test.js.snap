// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelButton renders 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "height": 36,
      "padding": 8,
      "width": 36,
    }
  }
>
  <ImageView
    imageProps={Object {}}
    source={null}
    style={
      Object {
        "alignSelf": "center",
        "height": 36,
        "padding": 8,
        "width": 36,
      }
    }
  />
</TouchableOpacity>
`;
